src/fft.jl

```diff
@@ -254,6 +254,23 @@ for P in (:cFFTWPlan, :rFFTWPlan, :r2rFFTWPlan) # complex, r2c/c2r, and r2r
     end
 end
 
+const DESTROY_QUEUE = Vector{FFTWPlan}(undef, 0)
+
+function destroy_queued_plans(t)
+    while length(DESTROY_QUEUE) > 0
+        destroy_plan(popfirst!(DESTROY_QUEUE)) # start with oldest
+    end
+    return nothing
+end
+
+const DESTROY_QUEUE_TIMER = Ref{Timer}(Timer(destroy_queued_plans, 0))
```

Comment:
This doesn't really seem reasonable to me; if there was a deadlock before, then this just introduces a race condition, no?  Also, a `Timer` with an interval of 0 seems like a spinloop, which troubles me.

Reply:
That timer has a delay of 0, so calls once  immediately. Interval is a kwarg that’s not used here.

My take on the original suggestion is to delay the destroying of plans until general fft planning has ended. But I don’t think the timer approach is right. Is there another event that could trigger destroying the queue, other than the finalizer?

Reply:
You could try to drain the queue manually after any operation returns. Since the timer will wait for `yield`, I think you can usually use a zero-length timeout (basically just an idle event, but we don't wrap those in libuv)

Reply:
I tried changing the 5 second timer to 0 seconds and the deadlock with the MWE returns.
1 second gives the same as 5 seconds, but 0.1 seconds results in deadlock. It seems too unstable to be considered a fix